### DISCUSSION TITLE
Propose moving color-related types from `rasterband.rs` into new `color.rs` file.

### DISCUSSION THREAD

**Discussion by metasim** [georust/gdal#316] [Ideas]
The `rasterband.rs` file covers an arguably broad set of concerns. I want to propose moving the types selected below into a new module.

<img width="462" alt="image" src="https://user-images.githubusercontent.com/131013/194196248-d54a6fdd-029a-4111-ab47-f01e2862461b.png">

In addition to this, two other questions might be worth pondering:

1. Should the refactored types be exposed as:
  a. `pub use color::*` in `raster` (to preserve binary compatibility, e.g. `gdal::raster::ColorTable`)?
  b. new paths (e.g. `gdal::raster::color::ColorTable`?
2. Should statistics-related items also be moved to a separate module?

**Comment by michaelkirk** [comment]
> a. pub use color::* in raster (to preserve binary compatibility, e.g. gdal::raster::ColorTable)?

If I'm understanding, the way I'd probably do this is:

1. move things to their new home.
2. re-export them from the legacy location so nothing breaks for people.
3. mark the legacy re-exports as deprecated so people can update their code when it's convenient for them.

Steps 1-3 can be done in a minor (non-breaking) release.

4. eventually (e.g. 6 months after release) delete the deprecated re-exports in a breaking release.

**Reply by metasim** [reply]
Great idea on deprecating the re-exports.
